api/middleware/notesMiddleware.js

```diff
@@ -0,0 +1,56 @@
+const Notes = require('../notes/noteModel');
+
+const checkNoteExists = async (req, res, next) => {
+  try {
+    const note = await Notes.findBy({ note_id: req.params.note_id });
+    if (!note) {
+      next({
+        status: 404,
+        message: 'Note with given ID does not exist',
+      });
+    } else {
+      //appending the already retrieved note to the req body to avoid multiple DB queries
+      req.body.retrievedNote = note;
+      next();
+    }
+  } catch (error) {
+    next(error);
+  }
+};
+
+const checkBodyIsComplete = (req, res, next) => {
+  if (
+    !req.body.content_type ||
+    !req.body.content ||
+    !req.body.level ||
+    !req.body.visible_to_admin ||
+    !req.body.visible_to_moderator ||
+    !req.body.visible_to_mentor ||
+    !req.body.profile_id_mentor ||
+    !req.body.profile_id_mentee
```

Comment:
Could this be simplified to something like `if (!body) {...}`?

Not going to stop this merge, but consider implementing simpler checks in the future and coming back to refactor this as soon as possible!